src/main/java/org/logstash/dissect/fields/AbstractField.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.dissect.fields;
+
+import com.logstash.Event;
+import org.logstash.dissect.Delimiter;
+import org.logstash.dissect.ValueResolver;
+
+import java.util.Map;
+
+public abstract class AbstractField implements Field {
```

Comment:
I missed this originally but stumbled upon it as I looked at your sub class implementations:

First of all, if you put something into a `HashMap`, you need to implement `equals()` and `hashCode`. At the moment your code works just because it is relying on object identity (meaning that it checks it is really at the same address in memory space, not logical equality). You should definitely implement equals and hashCode for all subclasses.

Secondly, using mutable classes as hash keys can get you into trouble very easily and it is hard to maintain. The reason is simple: You might not be able to find your key again because a property has changed that influenced the return value of equals/hashCode. :)